cmd/queue/main.go

```diff
@@ -201,8 +205,57 @@ func createAdminHandlers() *http.ServeMux {
 	return mux
 }
 
+func probeQueueHealthPath(port string) error {
+	url := fmt.Sprintf("http://127.0.0.1:%s%s", port, queue.RequestQueueHealthPath)
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			// Do not use the cached connection
+			DisableKeepAlives: true,
+		},
+		Timeout: 500 * time.Millisecond,
+	}
+
+	var lastErr error
+
+	err := wait.PollImmediate(100*time.Millisecond, time.Second, func() (bool, error) {
+		var res *http.Response
+		res, lastErr = httpClient.Get(url)
+
+		if res == nil {
+			return false, nil
+		}
+
+		defer res.Body.Close()
+		return res.StatusCode == http.StatusOK, nil
+	})
+
+	if lastErr != nil {
+		return fmt.Errorf("failed to probe: %s", lastErr)
+	}
+
+	// An http.StatusOK was never returned during probing
+	if err != nil {
+		return errors.New("probe returned not ready")
+	}
+
+	return nil
+}
+
 func main() {
 	flag.Parse()
+
+	if *probe {
+		port := strconv.Itoa(networking.QueueAdminPort)
```

Comment:
This seems not needed. Can we not pass the port as an int to `probeQueueHealthPath` and use `%d` for its interpolation?

Reply:
👍